music-player/src/components/AddSongToPlaylist.tsx

```diff
@@ -0,0 +1,80 @@
+import { Modal, Button } from 'antd';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { Dispatch } from 'redux';
+
+import { State } from '../types';
+import { getPlaylists } from '../reducers';
+import { getAddSongToPlaylistAction } from '../actions';
+
+class AddSongToPlaylist extends React.Component<
+  {
+    playlistsNotIncludeSongIndex: string[];
+    realIndex: number;
+    dispatch: Dispatch;
+  },
+  { show: boolean }
+> {
+  constructor(props: any) {
+    super(props);
+    this.state = { show: false };
+  }
+
+  closeWindowFunction = () => this.setState({ show: false });
+
+  render = () => {
+    const { playlistsNotIncludeSongIndex, dispatch } = this.props;
+    return (
+      <span>
+        <Button
+          className="song-button add-song-button"
+          shape="circle"
+          icon="plus"
+          size="small"
+          onClick={() => this.setState({ show: true })}
+        />
+        <Modal
+          title="Add Song to Playlist"
+          visible={this.state.show}
+          onCancel={this.closeWindowFunction}
+          footer={[
+            <Button key="ok" type="primary" onClick={this.closeWindowFunction}>
+              OK
+            </Button>,
+          ]}
+        >
+          {!this.state.show
```

Comment:
This line is useless since the modal is only visible if `this.state.show`

Reply:
There was a bug, before the modal was closed it showed a list of all the songs for a millisecond (you saw this too).
The cause was that the modal renders again before the closing animation. So it actually visible for a very short time.